R/DiseasyModelOdeSeir.R

```diff
@@ -0,0 +1,1231 @@
+#'
+#'   Similarly, the number of age groups in the model can also be controlled to create structurally different models.
+#'
+#'   The model implements the following features:
+#'   - A scaling of infection risk based on season (via `DiseasySeason`)
+#'   - Contact matrices and activity scenarios (via `DiseasyActivity`)
+#'   - Waning of immunity (via `DiseasyImmunity`)
+#'   - Asymmetric cross-immunity interactions between variants (via `DiseasyVariant`)
+#'
+#'   See `vignette(diseasy-model-ode-seir)` for a detailed examples of how to use this model.
+#' @examplesIf rlang::is_installed(c("duckdb", "deSolve"))
+#'   # The model can be instantiated almost without arguments, but
+#'   # to illustrate its use, we configure a simple model:
+#'
+#'   # First, we add a observables modules with example data bundled
+#'   # with the package.
+#'   obs <- DiseasyObservables$new(
+#'     diseasystore = DiseasystoreSeirExample,
+#'     conn = DBI::dbConnect(duckdb::duckdb())
+#'   )
+#'
+#'   # The observables module also defines the time if interest via
+#'   # the `last_queryable_date` field. Data before this date are
+#'   # used to train the models, and predictions start on this date.
+#'   obs$set_last_queryable_date(as.Date("2020-02-29"))
+#'
+#'   # The example data uses a simple activity scenario for Denmark,
+#'   # which we replicate here
+#'   act <- DiseasyActivity$new(contact_basis = contact_basis$DK)
+#'   act$set_activity_units(dk_activity_units)
+#'   act$change_activity(date = as.Date("2020-01-01"), opening = "baseline")
+#'
+#'   # We create a default instance which has:
+#'   # * 1 age group (0+)
+#'   # * 1 variant
+#'   # * No season scaling
+#'   # * No activity scenarios
+#'   m <- DiseasyModelOdeSeir$new(
+#'     observables = obs,
+#'     activity = act,
+#'     disease_progression_rates = c("E" = 1 / 2, "I" = 1 / 4),
+#'     parameters = list("overall_infection_risk" = 0.025)
+#'   )
+#'
+#'   # We need the initial state for the system, which we infer from
+#'   # the incidence data (see `vignette("SEIR-initialisation")`).
+#'   # We compute this here from the example data which uses a 65 %
+#'   # change of testing when infected
+#'   incidence_data <- m$observables$get_observation(
+#'     observable = "n_positive",
+#'     start_date = obs$ds$min_start_date,
+#'     end_date = m$training_period$end
+#'   ) |>
+#'     dplyr::mutate(
+#'       "incidence" = .data$n_positive / (sum(act$contact_basis$population) * 0.65)
+#'     )
+#'
+#'   psi <- m$initialise_state_vector(incidence_data)
+#'
+#'   # The model now has a configured right-hand-side function that
+#'   # can be used to simulate the model in conjunction with deSolve.
+#'   sol <- deSolve::ode(
+#'     y = psi$initial_condition,
+#'     times = seq(from = 0, to = 100, by = 1),
+#'     func = m$rhs
+#'   )
+#'
+#'   # Extract the incidence outcome from the solution
+#'   time <- sol[, 1]
+#'   model_incidence <- sol[, 3] * 1 / 4
```

Comment:
måske m$disease_progression_rates$I istedet for 1/4

Reply:
Might be a good idea to do it programmatically as you suggest.